python/phonenumbers/phonenumberutil.py

```diff
@@ -2772,6 +2778,7 @@ def _set_italian_leading_zeros_for_phone_number(national_number, numobj):
 
 def parse(number, region=None, keep_raw_input=False,
           numobj=None, _check_region=True):
+    # type: (str, str, bool, **Any, bool) -> None
```

Comment:
Again, without knowing much about Python type annotations I'd expect the 4th parameter to have a type like `Optional[PhoneNumber]` &ndash; does that not work?

(BTW, the codebase has a general convention that a parameter called `numobj` is supposed to be a `PhoneNumber` object)

Reply:
I have just tested this under Python 2 and 3, and it does appear to work. Pardon my confusion on the backwards compatible aspects. I will amend the next PR with this as well.

Reply:
Cool, sounds good &ndash; thanks for doing this.

By the way, how are you checking that the type annotations work?  I'm thinking that I could/should add some type checking to the continuous integration GitHub actions.

Reply:
No problem, happy to contribute!

To be specific, what I am implementing is type _commenting_ rather than annotating. Both were introduced in Python 3, but annotating has not been backported to Python 2. Commenting is the suggested alternative to annotating if backwards compatibility needs to be maintained under the heading "Suggested syntax for Python 2.7 and straddling code" in PEP 484.
I've been validating by ensuring the examples I wrote run correctly and also that my IDE (PyCharm) correctly parses and code-completes the functions for which the type hints have been added.
Thinking about checking, I am sure there would be some way to implement that, but I'm not certain at this moment how. Since I have been digging into the documentation, I will give this some thought and raise a feature request if I can come up with something that works once this pull request is complete.

Reply:
I've got a hack to do `mypy` checking in #203; it will be interesting to see how it combines with this PR&hellip;

Reply:
I don't yet understand the failures in https://github.com/daviddrysdale/python-phonenumbers/runs/3363632239?check_suite_focus=true so I think I've got some more reading to do.